library/core/src/ptr/const_ptr.rs

```diff
@@ -106,6 +106,75 @@ impl<T: ?Sized> *const T {
         bits as Self
     }
 
+    /// Gets the "address" portion of the pointer.
+    ///
+    /// This is equivalent to `self as usize`, which semantically discards
+    /// *provenance* and *address-space* information. To properly restore that information,
+    /// use [`with_addr`][pointer::with_addr] or [`map_addr`][pointer::map_addr].
+    ///
+    /// On most platforms this information isn't represented at runtime, and so the loss
+    /// of information is "only" semantic. On more complicated platforms like miri, CHERI,
+    /// and segmented architectures, this may result in an actual change of representation
+    /// and the loss of information.
```

Comment:
Miri isn't a platform, it's more like a codegen backend. Also I am not sure what '"only" semantic' means.^^

The important thing to get across is: Rust programs behave as if they were running on an Abstract Machine that has a bunch of fancy features real hardware does not have. Those fancy features are exploited by the compiler for its optimizations. One of those fancy features is 'provenance'. So when deciding whether a Rust program has Undefined Behavior, provenance is as real as everything else, and should be treated as such by programmers.

That should probably go somewhere central though, and be referenced from this method.